Add page on KIC vs Hybrid mode for Kubernetes (#7480)
Kong/docs.konghq.com: app/_src/gateway/production/deployment-topologies/kubernetes.md

---
title: Kubernetes Topologies
content_type: explanation
---

> Running {{ site.base_gateway }} on Kubernetes is a common deployment pattern. When running on Kubernetes, there are two configuration options available:

Reviewer: ```suggestion
Running {{ site.base_gateway }} on Kubernetes is a common deployment pattern. When running on Kubernetes, there are two deployment modes available:
```

> * [Hybrid mode](#hybrid-mode), with a CP (either on-prem or via Konnect) and multiple DPs

Reviewer: ```suggestion
* [Hybrid mode](#hybrid-mode): with an on-prem or Konnect based Control Plane (CP) and multiple Data Planes (DP).
```

Can we link to a definition for CP and DP?

> * [DB-less mode](/gateway/{{ page.release }}/production/deployment-topologies/db-less-and-declarative-config/), configured using CRDs and the {{ site.kic_product_name }}

Reviewer: ```suggestion
* [DB-less mode](/gateway/{{ page.release }}/production/deployment-topologies/db-less-and-declarative-config/): Uses Kubernetes Custom Resources with the {{ site.kic_product_name }} to configure the gateway
```

> Running Kong in Hybrid mode is commonly referred to as "Kong _on_ Kubernetes". Running Kong with {{ site.kic_product_name }} is commonly referred to as "Kong _for_ Kubernetes", as it provides a Kubernetes native way of configuring Kong entities using {{ site.kic_product_name }}.

Reviewer: I was asked not to use "Kong" in this way to mean KGW

Author: Understood. This is historical nomenclature so I'm comfortable keeping it for now

{:.note}
> You can also use {{ site.kic_product_name }} to configure a Hybrid mode deployment. This should only be used in a small set of circumstances. We recommend using Hybrid mode without KIC, or DB-less mode with KIC, unless you've been otherwise advised by a member of the Kong team.

## Hybrid mode

Deploying {{ site.base_gateway }} in [Hybrid mode](/gateway/{{ page.release }}/production/deployment-topologies/hybrid-mode/) uses Kubernetes as a runtime for your data planes. 

> Configuring Kong on Kubernetes is identical to deploying Kong running on a virtual machine or bare metal. The Data Planes connect to a Control Plane (Konnect, or an in-cluster control plane) and receive configuration from the CP. Configuration is managed using the UI, the Kong Admin API or via [deck](/deck/latest/).

Reviewer: ```suggestion
Configuring Kong on Kubernetes is identical to deploying Kong running on a virtual machine or bare metal. The DPs connect to a CP (Konnect, or in-cluster) to receive the configuration. Configuration is managed using the UI, the Kong Admin API or via [deck](/deck/latest/).
```

> We provide detailed [installation instructions](/gateway/{{ page.release }}/install/kubernetes/proxy/) for Hybrid mode, including ingress definitions for EKS, GKE and AKS.

Reviewer: ```suggestion
We provide detailed [installation instructions](/gateway/{{ page.release }}/install/kubernetes/proxy/) for Hybrid mode, including ingress definitions for popular managed Kubernetes services.
```

Author: I'm going to keep the specific providers here as those are the only ones we have docs for. Our audience will know the acronyms

## DB-less mode with {{ site.kic_product_name }}

> {{ site.kic_product_name }} provides a Kubernetes native way to configure {{ site.base_gateway }} using custom resource definitions (CRDs). In this deployment pattern, {{ site.base_gateway }} is deployed in DB-less mode, where the data plane configuration is held in memory.

Reviewer: ```suggestion
{{ site.kic_product_name }} provides a Kubernetes native way to configure {{ site.base_gateway }} using CRDs. In this deployment pattern, {{ site.base_gateway }} is deployed in DB-less mode, where the DP configuration is held in memory.
```

> Operators configure {{ site.base_gateway }} using standard CRDs such as `Ingress` and `HTTPRoute`, and {{ site.kic_product_name }} translates those resources in to Kong entities before sending a request to update the running data plane configurations.

Reviewer: ```suggestion
Operators configure {{ site.base_gateway }} using standard CRDs such as `Ingress` and `HTTPRoute`, and {{ site.kic_product_name }} translates those resources in to Kong entities before sending a request to update the running DP configurations.
```
